tests/v2/test_0449-merge-many-arrays-in-one-pass.py

```diff
@@ -3,108 +3,84 @@
-                    combined = np.concatenate([one, two, three, four])
-
-                    ak_combined = ak._v2.contents.NumpyArray(one).mergemany(
-                        [
-                            ak._v2.contents.NumpyArray(two),
-                            ak._v2.contents.NumpyArray(three),
-                            ak._v2.contents.NumpyArray(four),
-                        ]
-                    )
-
-                    assert to_list(ak_combined) == combined.tolist()
-                    assert ak_combined.dtype == combined.dtype
-
-                    assert (
-                        ak._v2.contents.NumpyArray(one)
-                        .typetracer.mergemany(
-                            [
-                                ak._v2.contents.NumpyArray(two),
-                                ak._v2.contents.NumpyArray(three),
-                                ak._v2.contents.NumpyArray(four),
-                            ]
-                        )
-                        .form
-                        == ak._v2.contents.NumpyArray(one)
-                        .mergemany(
-                            [
-                                ak._v2.contents.NumpyArray(two),
-                                ak._v2.contents.NumpyArray(three),
-                                ak._v2.contents.NumpyArray(four),
-                            ]
-                        )
-                        .form
-                    )
-
-                    ak_combined = ak._v2.contents.NumpyArray(one).mergemany(
-                        [
-                            ak._v2.contents.NumpyArray(two),
-                            ak._v2.contents.EmptyArray(),
-                            ak._v2.contents.NumpyArray(four),
-                        ]
-                    )
-
-                    assert to_list(ak_combined) == combined.tolist()
-                    assert ak_combined.dtype == np.concatenate([one, two, four]).dtype
-
-                    assert (
-                        ak._v2.contents.NumpyArray(one)
-                        .typetracer.mergemany(
-                            [
-                                ak._v2.contents.NumpyArray(two),
-                                ak._v2.contents.EmptyArray(),
-                                ak._v2.contents.NumpyArray(four),
-                            ]
-                        )
-                        .form
-                        == ak._v2.contents.NumpyArray(one)
-                        .mergemany(
-                            [
-                                ak._v2.contents.NumpyArray(two),
-                                ak._v2.contents.EmptyArray(),
-                                ak._v2.contents.NumpyArray(four),
-                            ]
-                        )
-                        .form
-                    )
+    for dtype1, dtype2, dtype3, dtype4 in itertools.product(
+        ("i1", "i2", "i4", "i8", "u1", "u2", "u4", "u8", "f4", "f8", "?"), repeat=4
+    ):
```

Comment:
Fine. It's hard for me to check because the un-indent is a diff: is the content of this loop unchanged?

Reply:
Yes, it just moves the four nested loops into a single product loop.